CI: clean up Chromedriver and references from repo.

worldomonation opened the pull request:
#### Changes proposed in this Pull Request

This PR removes dependencies and references to Chromedriver from the repository.

Key changes:
- removal of references to `CHROMEDRIVER_SKIP_DOWNLOAD` and `CHROMEDRIVER_SKIP_DOWNLOAD` env vars;
- removal of dependency on `chromedriver@88.x.x` from yarn;
- discontinue installation of Chrome in the docker image.

We no longer use Selenium in this repository as evidenced by https://github.com/Automattic/wp-calypso/pull/63108. Since Playwright manages its own browser installations, it is safe to remove Chrome and Chromedriver from the generated docker image.

#### Testing instructions

Ensure the following:
  - [ ] Gutenberg E2E (desktop)
  - [ ] Gutenberg E2E (mobile)
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)
  - [ ] Pre-Release E2E
  - [x] P2 E2E

Related to https://github.com/Automattic/wp-calypso/pull/63108

worldomonation committed:
Update the ci-e2e docker image:
- remove duplicate package statements
- discontinue installation of the Chrome browser package

noahtallen commented:
Why did we need it before? Any risks?

worldomonation commented:
> Why did we need it before? Any risks?

Ah, I should have clarified, sorry - Chrome was installed solely for Selenium tests. 
Now that Selenium has been entirely removed from the repository and no remaining jobs exist, I feel it is safe to remove it.

> the branch-level docker image build will utilize the daily base image build

My understanding of our pipelines is that:

- Docker base image is generated once a day using the Dockerfile.base that is in `trunk` at that time.
- once the build is complete, the images are uploaded and tags updated.
- subsequent PRs cut from `trunk` will use the new image, while older PRs will continue to use the older images.

Essentially, my understanding is that changes to `Dockerfile.base` will eventually propagate out to all branches. Is this not the case?

noahtallen commented:
Yes, that's the case! I just meant that we would have to take extra manual steps to test the best build in this branch before merging

noahtallen reviewed (approved):
The change makes sense to me, given it's not needed anywhere!

worldomonation committed:
Remove chromedriver wholesale:
- remove dependency on chromedriver@88
- remove referenes to chromedriver & related environment variables
- remove calypso-doctor > chromedriver

worldomonation committed:
Missed two files

worldomonation commented:
@noahtallen Just a heads up that I expanded the scope of this PR quite a bit.

Instead of only discontinuing the installation of Chrome browser at the Dockerfile level, the revised PR:
- removes dependency on `chromedriver` package from the repository;
- removes chromedriver-related environment variables such as `CHROMEDRIVER_SKIP_DOWNLOAD` and `DETECT_CHROMEDRIVER_VERSION`;
- updates documentation to remove references to chromedriver;
- removes the chromedriver part of calypso-doctor.

I'm putting the merge of this PR on hold since the change is rather extensive and see if you'd like to have a look again.

noahtallen commented on test/e2e/docs/troubleshooting.md:
````diff
@@ -66,7 +66,6 @@ Solution:
 
 ```
 PUPPETEER_SKIP_DOWNLOAD=true
````
Are we able to remove this too yet?

worldomonation replied:
I don't actually know where Puppeteer is used in `wp-calypso` and in my local environment I've always sidestepped the issue by setting that env var to `true`.

I'll do a quick check and if it's not too big of a change, will include it in this PR.

noahtallen replied:
I think I might remember. It could be pulled in via `@wordpress/scripts` or some of the wordpress monorepo helpers

worldomonation replied:
Searched for `puppeteer` and the only references I can find are:
- docs
- in yarn.lock file

Since we no longer run the Visual Regression suite (which used puppeteer) I think it might be safe to remove it?

worldomonation replied:
(additionally, the version of Puppeteer that's in `yarn.lock` is too old to support arm64 macOS)

worldomonation replied:
In any case, let's leave it for now and we can revisit Puppeteer in another patch.

noahtallen reviewed (approved):
Nice! These changes also look good to me, trusting that the e2e tests don't need it

noahtallen commented:
Is the failing mobile e2e related? it looked like it might possibly be, but I'm not sure

worldomonation commented:
> Is the failing mobile e2e related? it looked like it might possibly be, but I'm not sure

It shouldn't be - the `page.screenshot: Target crashed` error is observed intermittently and I am not sure why.

I do plan to upgrade the Playwright version soon, hopefully it will go away with that.
